### PR TITLE
Metricbeat : new metricset image for docker module 

### PR DESCRIPTION
**Pull request by douaejeouit**
Basic informations about the images available on the server

### CODE REVIEW

**Comment by karmi** [comment]
Hi @douaejeouit, we have found your signature in our records, but it seems like you have signed with a different e-mail than the one used in yout Git [commit](https://github.com/elastic/beats/pull/3467.patch). Can you please add both of these e-mails into your Github profile (they can be hidden), so we can match your e-mails to your Github profile?

**Comment by douaejeouit** [comment]
@karmi it's done!
Thanks

**Comment by douaejeouit** [comment]
@ruflin your review is welcome! 
thanks

**Comment by karmi** [comment]
@douaejeouit thanks!, but it keeps failing, have you added your Gmail to the Github profile? That should normally suffice to make that green, even when the commit has a different e-mail.

**Comment by douaejeouit** [comment]
@karmi that's weird 😢 I added both of my e-mails ..

**Comment by karmi** [comment]
Hi @douaejeouit, I've reached out via e-mail, ping me if you need assistance!

File: metricbeat/tests/system/test_docker.py [hunk 1]
```diff
@@ -191,6 +191,32 @@ def test_health_fields(self):
...
+        proc.check_kill_and_wait()
+
+        # Ensure no errors or warnings exist in the log.
+        log = self.get_log()
+        self.assertNotRegexpMatches(log.replace("WARN EXPERIMENTAL", ""), "ERR|WARN")
+
+        output = self.read_output_json()
+        evt = output[0]
+
+        evt = self.remove_labels(evt)
```
Comment (ruflin) [inline metricbeat/tests/system/test_docker.py]: As the image data set probably doesn't have fields, this line is not needed

File: metricbeat/module/docker/image/data.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+			"parent":  image.ParentID,
+		},
+		"created": image.Created,
+		"size": common.MapStr{
+			"regular": image.Size,
+			"virtual": image.VirtualSize,
+		},
+		"repoTags": image.RepoTags,
+	}
+	labels := docker.DeDotLabels(image.Labels)
```
Comment (ruflin) [inline metricbeat/module/docker/image/data.go]: Oh, I see it has labels. Ignore my comment below. But it seems the line with the labels somehow breaks the tests. See Travis.

File: metricbeat/module/docker/image/_meta/fields.yml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+- name: image
+  type: group
+  description: >
+    image
+  fields:
+    - name: example
+      type: keyword
+      description: >
+        Example field
```
Comment (ruflin) [inline metricbeat/module/docker/image/_meta/fields.yml]: This must be completed with the fields, otherwise the system test will not pass.

Comment (douaejeouit) [inline metricbeat/module/docker/image/_meta/fields.yml]: Ok, I'll complete it!

File: metricbeat/module/docker/image/data.go (hunk 2 above)
Comment (douaejeouit) [inline metricbeat/module/docker/image/data.go]: Yes, I'll check it!

**Comment by douaejeouit** [comment]
@ruflin Have you thought about the way we should document lists?

**Comment by ruflin** [comment]
@douaejeouit Which lists are you referring to? labels? My solution currently is to not document it (which is not great). Perhaps we could at least define an empty object: https://www.elastic.co/guide/en/elasticsearch/reference/current/object.html

**Comment by ruflin** [comment]
@douaejeouit I found that we use `dict-type: keyword` here which could become useful: https://github.com/elastic/beats/blob/master/winlogbeat/_meta/fields.yml#L40 But currently I think it does not work with our generation script.

**Comment by douaejeouit** [comment]
@ruflin , yes I am referring to labels as well as tags.
I did the same, I have not documented neither labels nor tags.
Defining an empty object sounds better!

**Comment by ruflin** [comment]
See https://github.com/elastic/beats/pull/3515

File: metricbeat/module/docker/image/_meta/data.json [hunk 4]
```diff
@@ -0,0 +1,19 @@
+{
```
Comment (ruflin) [inline metricbeat/module/docker/image/_meta/data.json]: Needs update but I can do that after merging.

**Comment by ruflin** [comment]
@douaejeouit Thanks for the contribution.

**Comment by douaejeouit** [comment]
With the greatest of pleasure @ruflin

**Pull request merged**